mindsdb/integrations/libs/vectordatabase_handler.py

```diff
@@ -301,6 +311,25 @@ def gen_hash(v):
         # id is string TODO is it ok?
         df[id_col] = df[id_col].apply(str)
 
+        # Handle document deletions first
+        doc_ids_to_delete = set()
+        for _, row in df.iterrows():
+            metadata = row.get(metadata_col, {})
+            if metadata and metadata.get('delete_existing') is True:
+                doc_id = metadata.get('original_doc_id')
+                if doc_id:
+                    doc_ids_to_delete.add(doc_id)
+
+        if doc_ids_to_delete:
+            # Delete existing chunks for these documents using metadata filter
+            conditions = [FilterCondition(
+                column=f"{metadata_col}->>'original_doc_id'",
```

Comment:
1. it is postgres specific syntax
I think it should filter by metadata key name: 
FilterCondition( column='original_row_id', ...)

2. maybe it should be original_row_id - it is id of the source